src/Data/Text/Utf8.hs

```diff
@@ -65,11 +127,49 @@ unpackUtf8 (Text u8data offset length) =
   in
     go offset length
 
+lengthUtf8 :: Text -> CodeUnitIndex
+lengthUtf8 (Text _ _ !length) = CodeUnitIndex length
+
+-- TODO: Make this more readable once we have text-2.0.
+--
+-- >  off                 off+len
+-- >   │                     │
+-- >   ▼                     ▼
+-- > ──┬─┬─┬─┬─┬─┬─┬─┬─┬─┬─┬─┬─┬──
+-- >  A│B│C│D│E│F│G│H│I│J│K│L│M│N
+-- > ──┴─┴─┴─┴─┴─┴─┴─┴─┴─┴─┴─┴─┴──
+-- >       ▲           ▲
+-- >       │           │
+-- >  off+begin   off+begin+length
+--
+-- Visualizes:
+--
+-- > unsafeCutUtf8 2 6 "BCDEFGHIJKL" == ("BC", "JKL")
+unsafeCutUtf8 :: CodeUnitIndex -> CodeUnitIndex -> Text -> (Text, Text)
```

Comment:
Can you still add the actual purpose of this function to the docstring, it was a bit hard to guess just based on the drawing. I.e., just copying the descriptions from the Utf16 variants.